### DISCUSSION TITLE
`VerticalAxis.ItemPlacer` returns different steps for positive and negative values

### DISCUSSION THREAD

**Discussion by mak1nt0sh** [patrykandpatrick/vico#882] [Ideas]
### How to reproduce

Create a line chart with both positive and negative values.

### Observed behavior

Using `VerticalAxis.ItemPlacer.step()` you get:

<img width="416" alt="Screenshot 2024-09-18 at 09 14 57" src="https://github.com/user-attachments/assets/5f5ea742-68be-41e8-8b34-3db35f34e949">

Using `VerticalAxis.ItemPlacer.count()` you get:

<img width="414" alt="Screenshot 2024-09-18 at 09 15 45" src="https://github.com/user-attachments/assets/6abfeb43-431f-4bac-9732-aad128b60622">

### Expected behavior

Steps should be the same for positive/negative

<img width="404" alt="Screenshot 2024-09-18 at 09 19 35" src="https://github.com/user-attachments/assets/7a39d374-2286-40de-8f37-73dadda18129">

### Vico version(s)

2.0.0-alpha.28

### Android version(s)

Any

### Additional information

Using manual step value works correctly.
`itemPlacer = remember { VerticalAxis.ItemPlacer.step(step = { 5000.00 }) },`

<img width="416" alt="Screenshot 2024-09-18 at 09 48 28" src="https://github.com/user-attachments/assets/3904840d-eb59-48a4-b8f0-72e38a9d5a38">

**Comment by Gowsky** [comment]
Hi @mak1nt0sh. It works this way by design so that the minimum, maximum, and 0 _y_ axis values are all labeled. As you noted, you can override this behavior with `step` by specifying your own step.

However, we can add APIs that use the same step at the top and bottom even when the step/count is determined automatically. So I'm converting this to a discussion as an "idea".

**Reply by mak1nt0sh** [reply]
I think it should use the same steps for both top and bottom as it is more intuitive when using the automatic api's. I was migrating from MPAndroid chart and was surprised that the steps are different. Looking at Cartesian charts in general, the ticks/labels are always displayed by the same steps and are read more clearly. It feels more like a workaround to display the zero values than a solution.

**Reply by Gowsky** [reply]
> I think it should use the same steps for both top and bottom as it is more intuitive when using the automatic api's.

As I mentioned, we'll add a built-in feature that produces such behavior. There will be a single, automatically calculated step for both top and bottom. We'll most likely make this the default.

It's already possible to achieve your desired behavior. You can do so by:

- Specifying the step explicitly, as we discussed.
- Creating your own `ItemPlacer` that uses a single automatically calculated step instead of two. You can base it on `DefaultVerticalAxisItemPlacer`, in which case only some minor modifications will be required.

> It feels more like a workaround to display the zero values than a solution.

It's not a workaround for anything, and it doesn't try to be a solution to anything. There's no problem that it attempts to overcome. It's just the behavior that was chosen. The result that it gives may not be the one that's desired in most cases, which is why the single-step behavior will most likely be made the default.

Again, you're not forced to rely on the two-step behavior. As mentioned above, it's already entirely possible to achieve the result you're looking for, including with an automatic step.

**Comment by mak1nt0sh** [comment]
Any update on making steps for positive/negative axis values the same?

It looks weird without it:
<img width="331" alt="Screenshot 2024-12-11 at 15 03 17" src="https://github.com/user-attachments/assets/0c69dec7-2784-4bef-aca6-d9d3a841eaa3" />
